### ISSUE TITLE
Invalid redeclaration error naming a let variable the same as a class.

### ISSUE THREAD

**Issue by paulvanbrenk** [Microsoft/TypeScript#1821]
Code
```ts
	class let {
		f(): let {
			let set = new let();
			return set;
		}
	}

	class set {
	}
```
I get the following error on the first let which I don't expect; Cannot redeclare block-scoped variable 'set'.

**Comment by RyanCavanaugh** [comment]
Please don't put the `ES6` tag on random bugs -- we're trying to use that for big-ticket feature/suggestion items

**Comment by vladima** [comment]
@paulvanbrenk interesting, just tried on the latest bits from master.
Running `node built\local\tsc.js -t es6 ..\test.ts` where 'test.ts' contains repro code does not produce any errors

**Comment by mhegazy** [comment]
@paulvanbrenk please reopen if you still see the issue.
